lib/stanford/storage_object_validator.rb

```diff
@@ -0,0 +1,63 @@
+require 'moab'
+
+module Stanford
+  # druids are Stanford specific entities
+  class StorageObjectValidator < Moab::StorageObjectValidator
+    # error codes
+    EXPECTED_DATA_SUB_DIRS = ["content", "metadata"].freeze
+    IMPLICIT_DIRS = ['.', '..'].freeze # unlike Find.find, Dir.entries returns these
+    DATA_DIR_NAME = "data".freeze
+    EXPECTED_VERSION_SUB_DIRS = [DATA_DIR_NAME, "manifests"].freeze
+    MANIFEST_INVENTORY_PATH = 'manifests/manifestInventory.xml'.freeze
+    SIGNATURE_CATALOG_PATH = 'manifests/signatureCatalog.xml'.freeze
+
+    INCORRECT_DIR = 0
+    MISSING_DIR = 1
+    EXTRA_CHILD_DETECTED = 2
+    VERSION_DIR_BAD_FORMAT = 3
+    NO_SIGNATURE_CATALOG = 4
+    NO_MANIFEST_INVENTORY = 5
+    NO_XML_FILES = 6
+    VERSIONS_NOT_IN_ORDER = 7
+    FILES_IN_VERSION_DIR = 8
+    DRUID_MISMATCH = 9
+
+    ERROR_CODE_TO_MESSAGES = {
+      INCORRECT_DIR=> "Incorrect items in path",
+      MISSING_DIR => "Missing directory: %{addl}",
+      EXTRA_CHILD_DETECTED => "Unexpected item in path: %{addl}",
+      VERSION_DIR_BAD_FORMAT => "Version directory name not in 'v00xx' format",
+      FILES_IN_VERSION_DIR => "Top level should contain only sequential version directories. Also contains files: %{addl}",
+      NO_SIGNATURE_CATALOG => "Version: %{addl} Missing signatureCatalog.xml",
+      NO_MANIFEST_INVENTORY => "Version: %{addl} Missing manifestInventory.xml",
+      NO_XML_FILES => "Version: %{addl} Missing all required metadata files",
+      VERSIONS_NOT_IN_ORDER => "Should contain only sequential version directories. Current directories: %{addl}",
+      DRUID_MISMATCH => "manifestInventory object_id does not match druid"
+    }.freeze
+
+    def validation_errors
+      errors = []
+      errors.concat identify_druid
+      super if errors.empty?
+      errors
+    end
+
+    def identify_druid
+      druid_from_filepath == object_id_from_manifest_inventory ? [] : [result_hash(DRUID_MISMATCH)]
+    end
+
+    private
+
+    def result_hash(response_code, addl=nil)
+      { response_code => error_code_msg(response_code, addl) }
+    end
```

Comment:
are `result_hash` and `error_code_msg` private methods the same as in superclass?  If so, shouldn't need to redeclare them.